### PR TITLE
Auto Localization

### PR DESCRIPTION
**Pull request by dimonovdd**
### Description of Change ###
This request adds automatic localization of  Cancel and Done buttons.

### API Changes ###
Added `useLocalizedButtons` parameter to the constructor

```csharp
public DatePickerDialog(UIColor? textColor = null, UIColor? buttonColor = null, UIFont? font = null, NSLocale? locale = null,
            bool showCancelButton = true, bool useLocalizedButtons = false)
```

### CODE REVIEW

**Comment by alexsorokoletov** [comment]
@dimonovdd thanks for the PR!
It's a great idea to use system done/cancel buttons for this.

Can you provide some visuals how it's going to look like with these changes?

**Comment by dimonovdd** [comment]
I can send a screenshot tomorrow

**Comment by alexsorokoletov** [comment]
@dimonovdd sure, whenever you have a chance!

File: DT.iOS.DatePickerDialog/DatePickerDialog.cs [hunk 1]
```diff
@@ -38,40 +41,44 @@ public class DatePickerDialog : UIView
...
-        {
-            Show(title, doneButtonTitle: "Done", cancelButtonTitle: "Cancel", datePickerMode: UIDatePickerMode.DateAndTime, callback: callback, defaultDate: DateTime.Now, maximumDate: maximumDate, minimumDate: minimumDate, cancelCallback: cancelCallback);
-        }
+            => Show(title, doneButtonTitle: kDone, cancelButtonTitle: kCancel, datePickerMode: UIDatePickerMode.DateAndTime, callback: callback, defaultDate: DateTime.Now, maximumDate: maximumDate, minimumDate: minimumDate, cancelCallback: cancelCallback);
 
         public void Show(string title, Action<DateTime> callback, UIDatePickerMode datePickerMode = UIDatePickerMode.DateAndTime, Action? cancelCallback = null)
-        {
-            Show(title, doneButtonTitle: "Done", cancelButtonTitle: "Cancel", datePickerMode: datePickerMode, callback: callback, defaultDate: DateTime.Now, cancelCallback: cancelCallback);
-        }
+            => Show(title, doneButtonTitle: kDone, cancelButtonTitle: kCancel, datePickerMode: datePickerMode, callback: callback, defaultDate: DateTime.Now, cancelCallback: cancelCallback);
```
Comment (alexsorokoletov) [inline DT.iOS.DatePickerDialog/DatePickerDialog.cs:59]: Thanks for this, I did not want to do that syntax refactoring in the recent release (did not have much time to get it right) so this is definitely welcome!

Comment (dimonovdd) [reply DT.iOS.DatePickerDialog/DatePickerDialog.cs:59]: I did it out of habit😀

File: DT.iOS.DatePickerDialog/DatePickerDialog.cs [hunk 2]
```diff
@@ -269,6 +287,46 @@ private void AddButtonsToView(UIView container)
...
+            toolbar.Layer.CornerRadius = kCornerRadius;
+            toolbar.Layer.MasksToBounds = true;
+
+            var doneButton = new UIBarButtonItem(UIBarButtonSystemItem.Done, ButtonTapped)
+            {
+                Tag = kDoneButtonTag
+            };
+            var cancelButton = _showCancelButton ? new UIBarButtonItem(UIBarButtonSystemItem.Cancel, ButtonTapped) : null;
+            var spacer = new UIBarButtonItem(UIBarButtonSystemItem.FlexibleSpace);
+            spacer.TintColor = UIColor.Red;
```
Comment (alexsorokoletov) [inline DT.iOS.DatePickerDialog/DatePickerDialog.cs]: **Optional**: if I'm reading this right https://blog.prototypr.io/designing-a-dark-mode-for-your-ios-app-the-ultimate-guide-6b043303b941 iOS has different red colors for dark/light themes.
If so then I'd move that to Colors where all theming is to keep it more or less aligned with the original Swift project.

image

**Comment by alexsorokoletov** [comment]
This looks good!
Post some screenshots and we'll move forward!

File: DT.iOS.DatePickerDialog/DatePickerDialog.cs (hunk 2 above)
Comment (dimonovdd) [reply DT.iOS.DatePickerDialog/DatePickerDialog.cs]: This is a typo. This line was needed for debug

**Comment by dimonovdd** [comment]
@alexsorokoletov 

I would like to make a few more improvements. Should I create new PRs for this?

**Comment by alexsorokoletov** [comment]
> @alexsorokoletov
> 
> I would like to make a few more improvements. Should I create new PRs for this?

If it's along the same lines as this PR - keep it updated and ping me when you're done.
If it's something else (not related to this PR) - make a new one.

**Comment by dimonovdd** [comment]
I'll create new PRs when we're merge with this.

**Comment by dimonovdd** [comment]
I tried to do this without any differences from the original design. But the size of the buttons is used by the system and the "Done" button has a bold font.

| Done (NO) | Done (En) | Dark Mode (En) |
|:------:|:-----:|:-----:|
|Done|DoneEn|DarkEn|
| **DoneCancel (NO)** | **DoneCancel (En)** | **DoneCancel (Ru)** |
|DoneCancel|DoneCancelEn|DoneCancelRu|

**Comment by alexsorokoletov** [comment]
It looks great!

**Comment by alexsorokoletov** [comment]
published in the Nuget as 1.0.5.

**Pull request merged**